frontend/editor.js

```diff
@@ -103,6 +103,28 @@ export const empty_notebook_state = ({ notebook_id }) => ({
  */
 const without_path_entries = (state) => ({ ...state, path: default_path, shortpath: "" })
 
+/**
+ * Fetches the statefile (usually a async resource) in launch_params.statefile
+ * and makes it available for consuming by `pluto-editor`
+ * To add custom logic instead, see use Environment.js
+ *
+ * @param {import("./components/Editor.js").LaunchParameters} launch_params
+ * @param {{current: import("./components/Editor.js").EditorState}} initial_notebook_state_ref
+ * @param {Function} set_ready_for_editor
+ * @param {Function} set_statefile_download_progress
+ */
+const get_statefile =
+    // @ts-ignore
+    window?.pluto_injected_environment?.custom_get_statefile?.(read_Uint8Array_with_progress, without_path_entries, unpack) ??
+    (async (launch_params, initial_notebook_state_ref, set_ready_for_editor, set_statefile_download_progress) => {
+        const r = await fetch(new Request(launch_params.statefile, { integrity: launch_params.statefile_integrity ?? undefined }))
+        const data = await read_Uint8Array_with_progress(r, set_statefile_download_progress)
+        const state = without_path_entries(unpack(data))
+        console.log({ state })
+        initial_notebook_state_ref.current = state
+        set_ready_for_editor(true)
+    })
```

Comment:
I would like to remove `initial_notebook_state_ref` and `set_ready_for_editor` from the arguments of `get_statefile`, and just make `get_statefile` a function that returns `Promse<State>`. 

So I would like to change this to:

```js
const get_statefile = (launch_params, set_statefile_download_progress) => {
	const r = await fetch(new Request(launch_params.statefile, { integrity: launch_params.statefile_integrity ?? undefined }))
    const data = await read_Uint8Array_with_progress(r, set_statefile_download_progress)
    const state = without_path_entries(unpack(data))
	return state
}

// later...

	useEffect(() => {
        if (!ready_for_editor && static_preview) {
            get_statefile(launch_params, set_statefile_download_progress).then((state) => {
				console.log({state})
		        initial_notebook_state_ref.current = state
		        set_ready_for_editor(true)
			})
        }
    }, [ready_for_editor, static_preview, statefile])
```

Does that work with your hook?

Reply:
(I edited my comment with a more complete snippet)